Mu2eUtilities/src/MueXSpectrum.cc

```diff
@@ -0,0 +1,123 @@
+// Spectrum based on arXiv: 1110.2874
+// Mu2e includes
+#include "GlobalConstantsService/inc/GlobalConstantsHandle.hh"
+#include "GlobalConstantsService/inc/PhysicsParams.hh"
+#include "GlobalConstantsService/inc/ParticleDataTable.hh"
+#include "Mu2eUtilities/inc/MueXSpectrum.hh"
+#include "Mu2eUtilities/inc/RandomUnitSphere.hh"
+
+// Framework includes
+#include "cetlib/pow.h"
+
+// CLHEP includes
+#include "CLHEP/Random/RandFlat.h"
+
+// C++ includes
+#include <iostream>
+#include <cmath>
+
+//GSL includes
+#include "stdio.h"
+#include "gsl/gsl_integration.h"
+
+using namespace std;
+  
+namespace mu2e {
+ 
+  MueXSpectrum::MueXSpectrum(double maxEnergy, double bin, int RadCorrected) :
+    _bin (bin),
+    _spectrumType (RadCorrected) 
+  {
+    GlobalConstantsHandle<ParticleDataTable> pdt;
+    _par.me    = pdt->particle(PDGCode::e_minus ).ref().mass().value();
```

Comment:
It looks like all members of the _par structure are hardcoded except eMax (the electron mass can be hardcoded as well). I would either make them adjustable parameters controllable by fcl, or directly use these numbers in the definition of f. That will simplifies the code a lot and get rid of  "(MueXSpectrum::Params_t*) p"

Reply:
I've removed a few things - I think leaving it as is for now is the best option since we might want to try different spectra in the future and it makes it clearer what all the parameters are and where they are used.